libraries/AP_Math/vector3.cpp

```diff
@@ -20,7 +20,8 @@
 
 #include "AP_Math.h"
 
-#define HALF_SQRT_2 0.70710678118654757f
+static const float HALF_SQRT_2 = 0.70710678118654757f;
```

Comment:
I am assuming that PIXHAWK1 is a single precision FPU.
But why do you write values that can not be expressed in single precision?
If the purpose is to reduce the value to be dropped, I think it would be better to set the other 7 decimal places to double precision.
If you think that, I would like to change other sources.

Reply:
@muramura I'm just trusting the compiler to do something appropriate. There's no reason to drop the precision myself - if I only give the compiler 32-bits to store something in I'm sure it will work something out :-)

This was pretty much a drive-by change to add some typing here.  I'm starting to significantly dislike `#define`s for constants.

Reply:
@peterbarker san... 
I am glad to hear your thoughts. Thank you.